lib/shared/models/WeightEntry.dart

```diff
@@ -1,6 +1,23 @@
 class WeightEntry {
+  final String? id; // Optional, will be set after Firestore creation
   final DateTime timestamp;
   final double weight;
 
-  WeightEntry({required this.timestamp, required this.weight});
+  WeightEntry({this.id, required this.timestamp, required this.weight});
+
+  Map<String, dynamic> toMap() {
+    return {
+      'id': id,
+      'timestamp': timestamp.toIso8601String(),
+      'weight': weight,
+    };
+  }
+
+  factory WeightEntry.fromMap(Map<String, dynamic> map, {String? id}) {
```

Comment:
rename `toMap` and `fromMap` to `toFirebase` and `fromFirebase` in case we need to change the firebase structure and keep some other JSON interpretation for the client